### PR TITLE
MOTECH-1706 Javadoc for Message Campaign (scheduler+builder)

### PR DESCRIPTION
**Pull request by sebbrudzinski**

### CODE REVIEW

File: message-campaign/src/main/java/org/motechproject/messagecampaign/scheduler/CampaignSchedulerService.java [hunk 1]
```diff
@@ -81,10 +141,34 @@ public void rescheduleMessageJob(CampaignEnrollment enrollment, Campaign campaig
...
     protected abstract void scheduleMessageJob(CampaignEnrollment enrollment, C campaign, M message);
 
+    /**
+     * Returns the delivery time, for the given {@link CampaignMessage}, that belongs to the provided {@link CampaignEnrollment}.
+     * It will return the time from enrollment if provided. Otherwise it will take the default time from the message definition.
+     *
+     * @param enrollment the enrollment the message belongs to
+     * @param message the message to fetch the delivery time for
+     * @return delivery time for the message
+     * @throws CampaignEnrollmentException if the delivery time has been provided neither in the enrollment nor in the message definition
```
Comment (pgesek) [inline message-campaign/src/main/java/org/motechproject/messagecampaign/scheduler/CampaignSchedulerService.java:170]: has not been provider

File: message-campaign/src/main/java/org/motechproject/messagecampaign/scheduler/EndOfCampaignListener.java [hunk 2]
```diff
@@ -6,12 +6,14 @@
...
 import org.motechproject.messagecampaign.domain.campaign.CampaignEnrollment;
 import org.motechproject.messagecampaign.domain.campaign.CampaignEnrollmentStatus;
-import org.quartz.SchedulerException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+/**
+ * MOTECH listener, that handles the campaign completed events.
```
Comment (pgesek) [inline message-campaign/src/main/java/org/motechproject/messagecampaign/scheduler/EndOfCampaignListener.java]: remove comma

File: message-campaign/src/main/java/org/motechproject/messagecampaign/scheduler/CampaignSchedulerService.java (hunk 1 above)
Comment (sebbrudzinski) [inline message-campaign/src/main/java/org/motechproject/messagecampaign/scheduler/CampaignSchedulerService.java:170]: @pgesek Should one use negation with neither?

Comment (pgesek) [inline message-campaign/src/main/java/org/motechproject/messagecampaign/scheduler/CampaignSchedulerService.java:170]: my bad

**Comment by sebbrudzinski** [comment]
Thanks for the comments, I've incorporated the changes.

**Pull request merged**